data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -192,13 +196,10 @@ protected void markRowDeleted(T item) {
   }
 
   public Predicate<T> eqDeletedRowFilter() {
-    if (eqDeleteRows == null) {
-      eqDeleteRows = applyEqDeletes().stream()
-          .map(Predicate::negate)
-          .reduce(Predicate::and)
-          .orElse(t -> true);
-    }
-    return eqDeleteRows;
+    return applyEqDeletes().stream()
+            .map(Predicate::negate)
+            .reduce(Predicate::and)
+            .orElse(t -> true);
```

Comment:
Please fix indentation. Continuation indents are 4 spaces. Rolling back this change should fix it.